crates/worker/src/partition/mod.rs

```diff
@@ -264,8 +264,24 @@ where
 {
     #[instrument(level = "error", skip_all, fields(partition_id = %self.partition_id, is_leader = tracing::field::Empty))]
     pub async fn run(mut self) -> anyhow::Result<()> {
-        info!("Starting the partition processor");
-        let res = self.run_inner().await;
+        info!("Starting the partition processor.");
+
+        let res = tokio::select! {
+            res = self.run_inner() => {
+                match res.as_ref() {
+                    Ok(_) => warn!("Shutting partition processor down because it stopped unexpectedly."),
+                    Err(err) => warn!("Shutting partition processor down because it failed: {err}"),
```

Comment:
Should these be `error!` or is there a case where this is okay to happen?

Reply:
It can happen if we fail to find the tail, for example. In this case, it's up to the `PPM` to handle this situation. Right now it would stop the PP and wait for further instructions from the CC. But maybe `warn` is too high since it can also happen when shutting down. The PPM would have the context to distinguish a shut down case from an unexpected termination.

Reply:
Is it possible to catch the shutdown error here?

Reply:
Yes, we can filter this condition out and keep the warning for the others.

Reply:
Ok, it is not so trivial because `err` is `anyhow::Error` and there are various places where we convert from other error types. Given that `ShutdownError` should only occur when shutting the system down, I would make this a follow-up for now.